gossipd/gossipd.c

```diff
@@ -1829,10 +1829,16 @@ static void gossip_send_keepalive_update(struct daemon *daemon,
 static void gossip_refresh_network(struct daemon *daemon)
 {
 	u64 now = gossip_time_now(daemon->rstate).ts.tv_sec;
-	/* Anything below this highwater mark could be pruned if not refreshed */
-	s64 highwater = now - daemon->rstate->prune_timeout / 2;
+	s64 highwater;
 	struct node *n;
 
+	/* For DEVELOPER testing, this can be set really short; otherwise, we
+	 * set it to 1 day before deadline. */
+	if (daemon->rstate->prune_timeout < 24*3600)
+		highwater = now - daemon->rstate->prune_timeout / 2;
+	else
+		highwater = now - (daemon->rstate->prune_timeout - 24*3600);
```

Comment:
Does it matter?  There's no danger in being pruned, as we'll come back and get unpruned?  And it would take both sides to be offline to be pruned like this.